packages/firestore/src/core/event_manager.ts

```diff
@@ -106,15 +106,29 @@ export class EventManager {
     }
   }
 
-  onChange(viewSnaps: ViewSnapshot[]): void {
-    for (const viewSnap of viewSnaps) {
-      const query = viewSnap.query;
-      const queryInfo = this.queries.get(query);
-      if (queryInfo) {
+  onChange(changes: {
```

Comment:
I pushed a commit that addresses this. I thought I could get away with this, but it turns out you are pretty good at catching up after vacation. Thanks for pointing out that this can be combined with the error handler, which makes the case for an interface even stronger.

For now, I left "subscribing" to the events untouched. EventManager still calls into SyncEngine`s subscribe method. I am torn on whether we should assign this in `firestore_client` since EventManager needs access to SyncEngine regardless.